spec/03-plugins/11-basic-auth/02-api_spec.lua

```diff
@@ -301,4 +302,104 @@ describe("Plugin: basic-auth (API)", function()
       end)
     end)
   end)
+  describe("/basic-auths", function()
+    local consumer2
+    describe("GET", function()
+      setup(function()
+        helpers.dao:truncate_table("basicauth_credentials")
+        assert(helpers.dao.basicauth_credentials:insert {
+          consumer_id = consumer.id,
+          username = "bob"
+        })
+        consumer2 = assert(helpers.dao.consumers:insert {
+          username = "bob-the-buidler"
+        })
+        assert(helpers.dao.basicauth_credentials:insert {
+          consumer_id = consumer2.id,
+          username = "bob-the-buidler"
+        })
+      end)
+      it("retrieve all the basic-auths", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths/"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(2, #json.data)
+        assert.equal(2, json.total)
+      end)
+      it("retrieve all the basic-auths", function()
+        local res = assert(admin_client:send {
+          method = "GET",
+          path = "/basic-auths"
+        })
+        local body = assert.res_status(200, res)
+        local json = cjson.decode(body)
+        assert.is_table(json.data)
+        assert.equal(2, #json.data)
+        assert.equal(2, json.total)
+      end)
```

Comment:
Will you also add a test case for pagination, like the one I added to `/key-auths/`? You can see it here: https://github.com/Kong/kong/commit/18f3a0939396c69a2554b28551d0a853581ef2d1#diff-a053792566ffb587af457e08fd34ec14R361

Reply:
Done